src/syscalls.py

```diff
@@ -7,16 +7,17 @@ class BaseSyscall(object):
     """
 
     # Take **kwargs and ignore to make life easier on RegularSyscall.
-    def __init__(self, x86=None, x64=None, **kwargs):
+    def __init__(self, x86=None, x64=None, generic=None, **kwargs):
         assert x86 or x64       # Must exist on one architecture.
         self.x86 = x86
         self.x64 = x64
+        self.generic = generic
         assert len(kwargs) is 0
 
 class RestartSyscall(BaseSyscall):
     """A special class for the restart_syscall syscall."""
-    def __init__(self, x86=None, x64=None):
-        BaseSyscall.__init__(self, x86=x86, x64=x64)
+    def __init__(self, x86=None, x64=None, generic=None):
+        BaseSyscall.__init__(self, x86=x86, x64=x64, generic=None)
```

Comment:
Shouldn't this pass through?

Reply:
oops, I triple checked the numbers, but forgot about the logic ;) Will fix

Reply:
#2553